frontend/src/app/shared/components/file-links/storage-file-list-item/storage-file-list-item.component.ts

```diff
@@ -0,0 +1,92 @@
+// -- copyright
+// OpenProject is an open source project management software.
+// Copyright (C) 2012-2022 the OpenProject GmbH
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License version 3.
+//
+// OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+// Copyright (C) 2006-2013 Jean-Philippe Lang
+// Copyright (C) 2010-2013 the ChiliProject Team
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License
+// as published by the Free Software Foundation; either version 2
+// of the License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+//
+// See COPYRIGHT and LICENSE files for more details.
+//++
+
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ElementRef,
+  EventEmitter,
+  Input,
+  OnInit, Output,
+  ViewChild,
+} from '@angular/core';
+
+import { TimezoneService } from 'core-app/core/datetime/timezone.service';
+import { IFileIcon } from 'core-app/shared/components/file-links/file-link-icons/icon-mappings';
+import {
+  getIconForMimeType,
+} from 'core-app/shared/components/file-links/file-link-icons/file-link-list-item-icon.factory';
+import { IStorageFile } from 'core-app/core/state/storage-files/storage-file.model';
+import { isDirectory } from 'core-app/shared/components/file-links/file-link-icons/file-icons.helper';
+import { PrincipalLike } from 'core-app/shared/components/principal/principal-types';
+
+@Component({
+  // eslint-disable-next-line @angular-eslint/component-selector
+  selector: '[op-storage-file-list-item]',
+  templateUrl: './storage-file-list-item.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class StorageFileListItemComponent implements OnInit {
+  @Input() public storageFile:IStorageFile;
+
+  @Input() public selected = false;
+
+  @Output() public changeSelection = new EventEmitter();
+
+  @ViewChild('avatar') avatar:ElementRef;
+
+  infoTimestampText:string;
+
+  fileLinkIcon:IFileIcon;
+
+  showDetails:boolean;
+
+  get principal():PrincipalLike {
+    return this.storageFile.createdByName
+      ? {
+        name: this.storageFile.createdByName,
+        href: '/external_users/1',
+      }
+      : {
+        name: 'Not Available',
+        href: '/placeholder_users/1',
+      };
```

Comment:
In ruby linting, ternary operators need to be on a single line or an if/else is to be used. I'd prefer that for readability.

Most of the values in here are static which looks of but admittedly, I haven't looked up the usage of it. What do the hrefs point to?

Reply:
I can understand that the ternary operator might look strange in this circumstance, it a personal preference to look at every step in the code as a function that takes input and created a value that is then passed around. Here the ternary operation creates an object containing a `name` and `href` from its inputs. I only use it for this purpose. I like it for it's simplicity. But if the linting is a hard requirement I'm fine changing it to `if/else`.

Secondly, the `href`s are hard coded strings to control the principal rendering. This will get removed on it's own somewhere in the future, when the principal renderer will be replaced.